source/common/modules/markdown-editor/plugins/render-codeblock-button.ts

```diff
@@ -0,0 +1,76 @@
+/**
+ * @ignore
+ * BEGIN HEADER
+ *
+ * Contains:        Codeblock copy button rendering plugin
+ * CVM-Role:        CodeMirror Plugin
+ * Maintainer:      Hendrik Erz
+ * License:         GNU GPL v3
+ *
+ * Description:     This plugin renders copy-to-clipboard button
+ *
+ * END HEADER
+ */
+
+import CodeMirror, { commands } from 'codemirror'
+
+const clipboard = window.clipboard
+/**
+   * Declare the  markdownRenderCodeBlockButton command
+   *
+   * @param   {CodeMirror.Editor}  cm  The CodeMirror instance
+*/
+;(commands as any).markdownRenderCodeBlockButton = function (cm: CodeMirror.Editor) {
+  cm.startOperation()
+
+  const codeBlockRE = /^(?:\s{0,3}`{3}|~{3}).*/
+  const lineCount = cm.lineCount()
+  let incodeblock = false
+  let codesblocks = new Array<string>()
+  let codeblock = ''
```

Comment:
Why don't you store the codeblock as an array? That would (a) make line 39 less awkward since you just had to push the line, would (b) save you from slicing in line 44, and (c) in line 71 you could simply write `codeblock.join('\n')` into the clipboard.